### DISCUSSION TITLE
Does Strelka scan files in SMTP traffic?

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#7781] [Unsupported Versions] [answered]
Hello!

Strelka successfully detected a malicious Excel file in SMB traffic (a week before our EDR did ... awesome!). For test purposes I sent this file again via e-mail to myself to see if Strelka will find the file in SMTP traffic as well. It didn't. Does Strelka do not scan SMTP traffic for files in the current configuration?

Cheers, Ben

**Comment by dougburks** [comment, marked as answer]
Strelka doesn't scan network traffic itself. From https://docs.securityonion.net/en/2.3/strelka.html:
> Depending on what options you choose in Setup, it may ask if you want to use [Zeek](https://docs.securityonion.net/en/2.3/zeek.html#zeek) or [Suricata](https://docs.securityonion.net/en/2.3/suricata.html#suricata) for metadata. Whichever engine you choose for metadata will then extract files from network traffic. Strelka then analyzes those files and they end up in /nsm/strelka/processed/.

**Reply by ben-sec** [reply]
Understood, than let me rephrase my question: Is it possible that Zeek extracts files successfully in SMB traffic, but not in SMTP traffic? By design or is that something you have to configure yourself?

**Reply by dougburks** [reply]
Zeek should be able to extract files from SMTP as well. You can review the Zeek file extraction config at `/opt/so/conf/zeek/policy/securityonion/file-extraction/`.

Are you sure the file was sent over cleartext SMTP?

Did Zeek log that SMTP connection in its smtp.log?

**Reply by palichx** [reply]
Hi, default zeek policy does not save smtp files. looks like "f?$total_bytes" is false for smtp file. (policy file /opt/so/saltstack/default/salt/zeek/policy/securityonion/file-extraction/extract.zeek)
Try to change condition to (!f?$total_bytes) && (f$source != "SMTP"). 
![image](https://user-images.githubusercontent.com/80513304/163579730-17af0f9c-cb82-4a85-96ac-50c8f018183e.png)

**Reply by dougburks** [reply]
I've confirmed that Zeek is not extracting files from some protocols due to the `total_bytes` check and I've created issue https://github.com/Security-Onion-Solutions/securityonion/issues/7829 to fix this.

**Reply by palichx** [reply]
May be need some additinional logic into zeek policy(protocol based extraction)?
For example:
I want to extract text/plain,text/html only from **SMTP** trraffic. But, if i enable it in /opt/so/saltstack/local/salt/zeek/fileextraction_defaults.yaml all text/plain,text/html will be extracted:
![image](https://user-images.githubusercontent.com/80513304/164414474-0e95fb98-f5c3-41a4-b975-d73410b8bcdc.png)

Sol for me: change extraction condition in file "/opt/so/saltstack/local/salt/zeek/policy/securityonion/file-extraction/extract.zeek" to 
`if ( !f$info?$md5 || ((!f?$total_bytes) && ((f$source != "SMTP") && (f$source != "HTTP") )) || f$missing_bytes > 0 || f$info$timedout || ((f$source == "HTTP") && ((f$info$mime_type == "text\/plain") || (f$info$mime_type == "text\/html") ))  ) `

**Reply by ben-sec** [reply]
You guys are great, thanks for looking into this one! In order to fight phishing attacks it would be awesome if file extraction for smtp would work flawlessly.
